File: src/main/java/OverrideExample.java
abstract class AbstractBase {
    abstract void parentFunction();
    final void notOverrideFunction(){};
}

public class OverrideExample extends AbstractBase{

    @Override
    void parentFunction() {

    }

//    @Override
//    void notOverrideFunction(){ // error
//
//    }
}

class TwiceOverrideExample extends OverrideExample{

}

File: src/main/kotlin/ch3/OverrideExample.kt
package ch3

abstract class AbstractBase{
    abstract fun parentFunction()
    fun notOverrideFunction(){}
}

open class AbstractOverrideExample() : AbstractBase(){
    override fun parentFunction() {

    }

//    override fun notOverrideFunction() { // error
//
//    }

}


class TwiceOverrideExample(): AbstractOverrideExample() {

}

